kornia/augmentation/random_generator/_2d/linear_illumination.py

```diff
@@ -0,0 +1,176 @@
+from kornia.core import Tensor
+from kornia.enhance.normalize import normalize_min_max
+from kornia.utils import _extract_device_dtype
+
+
+class LinearIlluminationGenerator(RandomGeneratorBase):
+    r"""Generates random 2D Linear illumination patterns for image augmentation.
+
+    Args:
+        gain: Range for the gain factor applied to the generated illumination.
+        sign: Range for the sign of the Linear distribution.
+
+    Returns:
+        A dictionary of parameters to be passed for transformation.
+            - gradient: : Generated 2D Linear illumination pattern with shape (B, C, H, W).
+
+    Note:
+        The generated random numbers are not reproducible across different devices and dtypes. By default,
+        the parameters will be generated on CPU. This can be changed by calling
+        ``self.set_rng_device_and_dtype(device="cuda", dtype=torch.float64)``.
+    """
+
+    def __init__(
+        self,
+        gain: tuple[float, float],
+        sign: tuple[float, float],
+    ) -> None:
+        super().__init__()
+        self.gain = gain
+        self.sign = sign
+
+    def __repr__(self) -> str:
+        r"""Return a string representation of the object."""
+        repr = f"gain={self.gain}, sign={self.sign}"
+        return repr
+
+    def make_samplers(self, device: torch.device, dtype: torch.dtype) -> None:
+        r"""Create samplers for generating random gaussian illumination parameters."""
+
+        gain = _range_bound(
+            self.gain,
+            "gain",
+        ).to(device, dtype)
+        self.gain_sampler = UniformDistribution(gain[0], gain[1], validate_args=False)
+
+        sign = _range_bound(self.sign, "sign", bounds=(-1.0, 1.0), center=0.0).to(device, dtype)
+        self.sign_sampler = UniformDistribution(sign[0], sign[1], validate_args=False)
+
+        self.directions_sampler = UniformDistribution(0, 4, validate_args=False)
+
+    def forward(self, batch_shape: tuple[int, ...], same_on_batch: bool = False) -> dict[str, Tensor]:
+        r"""Generate random 2D Gaussian illumination patterns."""
+        batch_size, channels, height, width = batch_shape
+        _common_param_check(batch_size, same_on_batch)
+        _device, _dtype = _extract_device_dtype([self.gain, self.sign])
+
+        gain_factor = _adapted_rsampling((batch_size, 1, 1, 1), self.gain_sampler, same_on_batch).to(
+            device=_device, dtype=_dtype
+        )
+        sign = torch.where(
+            _adapted_rsampling((batch_size, 1, 1, 1), self.sign_sampler, same_on_batch) >= 0.0,
+            torch.tensor(1),
+            torch.tensor(-1),
+        ).to(device=_device, dtype=_dtype)
+
+        directions = _adapted_rsampling((batch_size, 1, 1, 1), self.directions_sampler, same_on_batch).to(
+            device=_device,
+            dtype=torch.int8,
+        )
+
+        gradient = torch.zeros(batch_shape)
+        for _b in range(batch_size):
```

Comment:
why it's a loop over the batch?

Reply:
No, the batch and direction are independent. I have to create a loop for batch because linspace does not accept batch, is the only way that I thought. And direction in a simple random value between 0 and 4.

Reply:
maybe you can workaround this with a meshgrid ?

Reply:
I'm seeing `meshgrid `, and only accept 0D or 1D array as input, so we can't work with channels or batch. I'm not sure how we can do it.

https://discuss.pytorch.org/t/batched-meshgrid/157206
https://stackoverflow.com/questions/72782751/how-to-perform-torch-meshgrid-over-multiple-tensors-in-parallel

Reply:
may this do the trick? https://github.com/kornia/kornia/blob/886b9328ae4a1fff39acd953392de80dbc0fd468/kornia/utils/grid.py#L9

Reply:
i think so -- side comment, can we make this performance improvement in a follow up PR ? @vgilabert94 we plan to release early next week and would be great to have this patch in.